modules/notifications/index.ts

```diff
@@ -50,15 +51,34 @@ export default (): Window<Box<Child, Attribute>, unknown> => {
             hexpand: true,
             setup: (self) => {
                 Utils.merge(
-                    [notifs.bind('popups'), ignore.bind('value')],
-                    (notifications: Notification[], ignoredNotifs: string[]) => {
+                    [notifs.bind('popups'), ignore.bind('value'), showActionsOnHover.bind('value')],
+                    (notifications: Notification[], ignoredNotifs: string[], showActions: boolean) => {
                         const filteredNotifications = filterNotifications(notifications, ignoredNotifs);
 
                         return (self.children = filteredNotifications.slice(0, displayedTotal.value).map((notif) => {
+                            const actionsbox =
+                                notif.actions.length > 0
+                                    ? Widget.Revealer({
+                                          transition: 'slide_down',
+                                          reveal_child: showActions ? false : true,
+                                          child: Widget.EventBox({
+                                              child: Action(notif, notifs),
+                                          }),
+                                      })
+                                    : null;
+
                             return Widget.EventBox({
                                 on_secondary_click: () => {
                                     notifs.CloseNotification(notif.id);
                                 },
+                                on_hover() {
+                                    if (actionsbox && showActions) actionsbox.reveal_child = true;
+                                },
+                                on_hover_lost() {
+                                    if (actionsbox && showActions) actionsbox.reveal_child = false;
+
+                                    notif.dismiss();
```

Comment:
Why are we dismissing a notification on hover loss here?

Reply:
Let me remove that. I guess thats not needed